Core/include/Acts/Propagator/EigenStepper.hpp

```diff
@@ -47,12 +49,32 @@ template <typename bfield_t,
           typename auctioneer_t = detail::VoidAuctioneer>
 class EigenStepper {
  public:
-  /// Jacobian, Covariance and State defintions
   using Jacobian = BoundMatrix;
-  using Covariance = BoundSymMatrix;
-  using BoundState = std::tuple<BoundTrackParameters, Jacobian, double>;
+
+  /// Variant Jacobian:
+  /// free-bound, bound-bound, bound-free, free-free
+  using VariantJacobian = std::variant<FreeToBoundMatrix, BoundMatrix,
+                                       BoundToFreeMatrix, FreeMatrix>;
+
+  /// Variant Jacobian initial to global:
+  /// bound-free, free-free
+  using VariantJacobianToGlobal = std::variant<BoundToFreeMatrix, FreeMatrix>;
```

Comment:
If you start in free parameters then you don't need this matrix. What's the benefit of overallocating memory compared to an `std::optional` that carries the same information?

Reply:
It's true - I might change that to `std::optional`